src/passkey_child/passkey_child_common.c

```diff
@@ -561,73 +626,22 @@ authenticate(struct passkey_data *data)
 {
     TALLOC_CTX *tmp_ctx = NULL;
     fido_assert_t *assert = NULL;
-    fido_dev_info_t *dev_list = NULL;
     fido_dev_t *dev = NULL;
     struct pk_data_t pk_data = { 0 };
-    size_t dev_list_len = 0;
-    int count = 0;
+    int index = 0;
```

Comment:
The index is initialized here instead of select_authenticator.

Reply:
Fixed